var/spack/repos/builtin/packages/neuron/package.py

```diff
@@ -0,0 +1,240 @@
+    depends_on('pkg-config', type='build')
+    depends_on('python@2.6:', when='+python')
+    depends_on('mpi', when='+mpi')
+
+    # on bg-q mpi needs to be enabled : todo test this
+    conflicts('~mpi', when='platform=bgq')
+
+    # pgi compiler can't build static libraries
+    conflicts('%pgi', when='+static')
+
+    def url_for_version(self, version):
+        url = "http://www.neuron.yale.edu/ftp/neuron/versions/v{0}/nrn-{0}.tar.gz"
+        return url.format(version, version)
+
+    def patch(self):
+        # aclocal need complete include path especially on os x
+        pkgconf_inc = '-I %s/share/aclocal/' % (self.spec['pkg-config'].prefix)
+        libtool_inc = '-I %s/share/aclocal/' % (self.spec['libtool'].prefix)
+        newpath = 'aclocal -I m4 %s %s' % (pkgconf_inc, libtool_inc)
+        filter_file(r'aclocal -I m4', r'%s' % newpath, "build.sh")
+
+    def get_arch_options(self, spec):
+        options = []
+
+        if 'bgq' in self.spec.architecture:
+            options.extend(['--enable-bluegeneQ',
+                            '--host=powerpc64',
+                            '--without-memacs'])
+
+        if 'cray' in self.spec.architecture:
+            options.extend(['--without-memacs',
+                            '--without-nmodl',
+                            'cross_compiling=yes'])
+
+            # TODO: on cray systems we get an error while linking even if
+            # we use cc wrapper. Hence add explicitly add mpi library
+            if spec.satisfies('+mpi'):
+                options.append('LIBS=-lmpich')
+
+            # TODO: -pthread is not a valid pthread option for cray compiler.
+            # for now disable use of pthread with cray compiler.
+            if spec.satisfies('%cce'):
+                options.append('use_pthread=no')
+
+        return options
+
+    def get_opt_flags(self):
+        if 'bgq' in self.spec.architecture:
+            return '-O3 -qtune=qp -qarch=qp -q64 -qstrict -qnohot -g'
+        else:
+            return '-O2 -g'
+
+    def get_arch_dir(self):
+        if 'bgq' in self.spec.architecture:
+            arch = 'powerpc64'
+        elif 'cray' in self.spec.architecture:
+            arch = 'x86_64'
+        else:
+            arch = self.spec.architecture.target
+
+        return arch
+
+    def get_python_options(self, spec):
+        options = []
+
+        if spec.satisfies('+python'):
+            py_prefix = spec['python'].prefix
+            py_version = 'python{0}'.format(spec['python'].version.up_to(2))
+            python_exec = '%s/bin/%s' % (py_prefix, py_version)
+
+            options.append('--with-nrnpython=%s' % python_exec)
+            options.append('--disable-pysetup')
+
+            if spec.satisfies('+cross-compile'):
+                # on cross compile builds we have to provide
+                # PYINCDIR, PYLIB, PYLIBDIR etc
+
+                py_lib = spec['python'].prefix.lib
+                py_inc = '%s/include/%s' % (py_prefix, py_version)
```

Comment:
@adamjstewart : thanks for pointer!

In the boost it is used as `spec['python'].libs[0]`. But I am seeing an error:

```
==> Executing phase : 'install'
==> '/usr/local/Cellar/python3/3.6.2/Frameworks/Python.framework/Versions/3.6/bin/python3.6' '-c' 'from distutils.sysconfig import get_config_h_filename; print(get_config_h_filename())'
---> /usr/local/Cellar/python3/3.6.2/Frameworks/Python.framework/Versions/3.6/include/python3.6m
==> '/usr/local/Cellar/python3/3.6.2/Frameworks/Python.framework/Versions/3.6/bin/python3.6' '-c' 'from distutils.sysconfig import get_config_var; print(get_config_var('"'"'LIBDIR'"'"'))'
==> '/usr/local/Cellar/python3/3.6.2/Frameworks/Python.framework/Versions/3.6/bin/python3.6' '-c' 'from distutils.sysconfig import get_config_var; print(get_config_var('"'"'PYTHONFRAMEWORKPREFIX'"'"'))'
==> '/usr/local/Cellar/python3/3.6.2/Frameworks/Python.framework/Versions/3.6/bin/python3.6' '-c' 'from distutils.sysconfig import get_config_var; print(get_config_var('"'"'LIBRARY'"'"'))'
==> Error: RuntimeError: Unable to locate python libraries in /usr/local/opt/python3/Frameworks/Python.framework/Versions/3.6/lib
```

The `lib` folder has:

```
$ ll /usr/local/opt/python3/Frameworks/Python.framework/Versions/3.6/lib/
total 16
drwxr-xr-x    6 kumbhar  admin   204 Jul 17 05:45 .
drwxr-xr-x    9 kumbhar  admin   306 Jul 17 05:45 ..
lrwxr-xr-x    1 kumbhar  admin     9 Jul 17 05:45 libpython3.6.dylib -> ../Python
lrwxr-xr-x    1 kumbhar  admin     9 Jul 17 05:45 libpython3.6m.dylib -> ../Python
drwxr-xr-x    5 kumbhar  admin   170 Aug 14 20:41 pkgconfig
drwxr-xr-x  205 kumbhar  admin  6970 Aug 14 20:41 python3.6
```

Also if I have:
```
l = LibraryList(['/usr/local/Cellar/python3/3.6.2/Frameworks/Python.framework/Versions/3.6/lib/libpython3.6.dylib'])
```

I expected `l.ld_flags` to be `-lpython3.6` but get `-lpython3`:

```
-L/usr/local/Cellar/python3/3.6.2/Frameworks/Python.framework/Versions/3.6/lib -lpython3
```

Anything obvious thing I am missing?